### ISSUE TITLE
Add Moodle group functionality

### ISSUE THREAD

**Issue by mei-cberry** [jrchamp/moodle-mod_zoom#443]
There is no functionality to schedule meetings for specific groups within Moodle. If a meeting is scheduled, it appears in the calendar at course level and so is visible to all users within the course. It would be very useful if there was an option to select a group when scheduling a meeting, and for the calendar event to be added at group level rather than course level in that scenario.

**Comment by jrchamp** [comment]
Before Moodle 2.8, there was experimental "group members only" functionality. mod_zoom added support for it in v1.1: 31db8fefebde4adffc36739b6d1528dee6944e72

From https://docs.moodle.org/28/en/Available_for_group_members_only
> "Available for group members only" was an experimental feature in earlier versions of Moodle and was removed in Moodle 2.8. The group restriction option in conditional activities now provides this functionality.

If the [Restrict access settings](https://docs.moodle.org/401/en/Restrict_access_settings) for groups are not showing up as part of the Moodle's standard additional module controls, then it may be due to a missing entry in our module's zoom_supports() function. Specifically, FEATURE_GROUPS is notably missing and may be the appropriate replacement for the deprecated/removed FEATURE_GROUPMEMBERSONLY that is currently present.

Once the restrict access setting is in place, testing will need to be done to make sure that the events generated are correctly only visible to group members (and instructors/admins of course) instead of the entire course.

**Comment by mei-cberry** [comment]
Restrict access can be used. I have checked and the events generated are only visible to group members, so that is working appropriately.
This is a satisfactory workaround, although it does mean that a separate instance must be created for each group. In the BigBlueButton meetings plugin for Moodle, there is groups functionality within the plugin so that a single instance involves a separate meeting for each group, so that was the sort of functionality I was looking for. However, the restrict access functionality does allow us to do what we need to do - thank you.

**Comment by jrchamp** [comment]
Thank you @mei-cberry! Glad to know that it is possible to get it working. The "one activity with separate Zoom meetings for each group" functionality would most like require a significant change to the underlying data structure. If that feature is desirable, it may be cleanest to open a separate issue. For now, I'll mark this specific issue as closed.
